packages/svelte/src/chat-context.svelte.ts

```diff
@@ -0,0 +1,23 @@
+import type { JSONValue, UIMessage } from '@ai-sdk/ui-utils';
+import { createContext, KeyedStore } from './utils.svelte.js';
+
+class ChatStore {
+  messages = $state<UIMessage[]>([]);
+  data = $state<JSONValue[]>();
+  status = $state<'submitted' | 'streaming' | 'ready' | 'error'>('ready');
+  error = $state<Error>();
+}
+
+export class KeyedChatStore extends KeyedStore<ChatStore> {
+  constructor(
+    value?: Iterable<readonly [string, ChatStore]> | null | undefined,
+  ) {
+    super(ChatStore, value);
+  }
+}
+
+export const {
+  hasContext: hasChatContext,
+  getContext: getChatContext,
+  setContext: setChatContext,
+} = createContext<KeyedChatStore>('Chat');
```

Comment:
is the use of ChatStore documented somewhere (in the ai sdk docs)?

Reply:
No, this is not exported (it's an internal implementation detail) -- however I just realized I need to document the context provider, will do that before merge